Sparrow.ps1

```diff
@@ -184,6 +187,21 @@ Function Get-UALData {
     #By default, it will show up as Consent_Operations_Export.csv       
     Export-UALData -ExportDir $ExportDir -UALInput $ConsentData -CsvName "Consent_Operations_Export" -WorkloadType "AAD"
 
+    #Searches for SAML token usage anomaly (UserAuthenticationValue of 16457) in the Unified Audit 
+    $domains = Get-MsolDomain | where {$_.Authentication -eq "Federated"}
+    $domainsToFlag = $domains | %{ Get-MsolDomainFederationSettings -DomainName $_.Name | where {$_.SupportsMfa -ne $true}}
+    If ($null -ne $domainsToFlag)
+    {
+        Write-Verbose "Searching for 16457 in UserLoggedIn and UserLoginFailed operations in the UAL."
+        $SAMLData = Search-UnifiedAuditLog -StartDate $StartDate -EndDate $EndDate -Operations "UserLoggedIn","UserLoginFailed" -ResultSize 5000 -FreeText "16457" | Select-Object -ExpandProperty AuditData | Convertfrom-Json
+        $FilteredSAMLData = $SAMLData | Where-Object {$_.UserId -like "*$domainsToFlag*"}
```

Comment:
when $domainsToFlag returns more than 1 domain in line 192, the -like will not work. 
You need to either make the user domain its own field and use the -in operator.
Something like 
$SAMLData | where-object {$_.UserId.Split('@')[1] -in $domainsToFlag}

Reply:
Feedback incorporated.